install.sh

```diff
@@ -0,0 +1,406 @@
+F_FG_CYAN="$(tput setaf 6)"
+F_BOLD="$(tput bold)"
+F_RESET="$(tput sgr0)"
+
+# Prints usage
+usage() {
+  cat <<EOF
+install.sh [options] [COMMAND]
+
+Installs the chs-dev CLI tool.
+
+OPTIONS
+-------
+
+-d <directory> - local directory to install CLI to (Defaults to ~/.chs-dev)
+-f - forces the installation - will clobber existing installation. Without the flag,
+    user will have to confirm reinstallation
+-l <loglevel> - sets the log level to set logging output, valid options are DEBUG,
+    INFO, WARN, ERROR (defaults to INFO)
+-S - when set does not create symlink to binary
+-s <path-to-symlink-loc> - Sets the path to the place to output the symlink to (Defaults to ~/.companies_house_config/bin)
+-v <version> - installs a specific version (Defaults to latest)
+-h print this message and exit
+
+ARGUMENTS
+---------
+
+COMMAND - Specifies the action to do, must be install or uninstall. Defaults to installl
+EOF
+}
+
+# Logs error and then exits process with error exit code
+panic() {
+  log ERROR "${1:?message required}" >&2
+  exit 1
+}
+
+# Logs messages to terminal and controls logging
+log() {
+  log_level="${1:?level required}"
+  log_message="${2:?message required}"
+
+  case "${LOGGING_LEVEL}" in
+  DEBUG) ;;
+  INFO)
+    if [ "${log_level}" = 'DEBUG' ]; then
+      return 0
+    fi
+    ;;
+  WARN*)
+    if [ "${log_level}" = 'DEBUG' ] || [ "${log_level}" = 'INFO' ]; then
+      return 0
+    fi
+    ;;
+  ERROR)
+    if [ ! "${log_level}" = "ERROR" ]; then
+      return 0
+    fi
+    ;;
+  esac
+
+  case "${log_level}" in
+  DEBUG) log_colour="" ;;
+  INFO) log_colour="${F_FG_GREEN}" ;;
+  WARN*) log_colour="${F_FG_YELLOW}" ;;
+  ERROR) log_colour="${F_FG_RED}" ;;
+  *) panic "Invalid log level: ${log_level}" ;;
+  esac
+
+  printf '[%s] - [%s] - %s\n' "${F_FG_CYAN}$(date +'%Y-%m-%dT%H:%M:%S')${F_RESET}" "${log_colour}${F_BOLD}${log_level}${F_RESET}" "${log_message}"
+}
+
+# Checks that the requested version is valid i.e. there is a release with the
+# name of the version supplied
+# RETURNS
+# 0  - version is valid
+# 1  - version is not valid
+validate_version() {
+  log DEBUG 'validating version'
+
+  {
+    jq -r '.[] | .name' "${releases_file}"
+    printf -- '%s' "$?" >"${temp_d}"/jq_status
+  } |
+    {
+      grep -q "${VERSION}"
+      printf -- '%s' "$?" >"${temp_d}"/grep_status
+    }
+
+  if ! grep -q '^0$' "${temp_d}"/jq_status "${temp_d}"/grep_status; then
+    log ERROR "Version ${VERSION} not found"
+    return 1
+  fi
+}
+
+# Sets the variable OS to the name of the operating system. Panics if the
+# operating system is unsupported by the CLI (i.e. missing compiled CLI
+# artifacts). Panics if not OSX or Linux compatible
+determine_operating_system() {
+  log DEBUG 'determining operating system'
+  uname_out="$(uname -a | tr '[:upper:]' '[:lower:]')"
+  case "${uname_out}" in
+  *microsoft*) OS="linux" ;; #WARNING: My v2 uses ubuntu 20.4 at the moment slightly different name may not always work
```

Comment:
lowercasing Microsoft fails to distinguish between WSL & WSL2. Also, why not call it wsl2? It's not strictly linux. You could nick the existing function from setup_lib.bash.

Reply:
It is all about working out which binary to download options are `linux` or `darwin`

Reply:
For context take a look at: https://github.com/companieshouse/chs-dev/releases/tag/0.1.4 at the available tarballs

Reply:
Therefore I don't think there is the need here to distinguish between the two unless they need different binaries?